### DISCUSSION TITLE
Snacks.dashboard.update() - help

### DISCUSSION THREAD

**Discussion by Praczet** [folke/snacks.nvim#128] [Q&A] [answered]
Hi,

The long story  short: 
I have a plugin (or at least I treat it as plugin) Little-TaskWarrior, which allows two things: Display tasks in dashboard, and run user command Task (which run cmd task)

I would like to be able refresh my tasks in the dashboard if open (this I can check)

Here is my function:

```lua
function M.refresh_dashboard()
	if 
           M.is_dashboard_open() 
           and M.get_dashboard_config 
           and type(M.get_dashboard_config) == "function" 
         then
		if Snacks and Snacks.dashboard and type(Snacks.dashboard.update) == "function" then
			-- local bufnr = vim.api.nvim_get_current_buf()
			-- vim.api.nvim_buf_delete(bufnr, { force = true })
			-- Snacks.dashboard.setup(M.get_dashboard_config())
			-- Snacks.dashboard.open(M.get_dashboard_config())
			Snacks.dashboard.update()
		else
			local bufnr = vim.api.nvim_get_current_buf()
			vim.api.nvim_buf_delete(bufnr, { force = true })
			local _dashboard = require("dashboard")
			_dashboard.setup(M.get_dashboard_config())
			_dashboard:instance()
		end
	end
end
```

My Snacks.dashboard setup looks like this:

```lua
local myFunc = require("my-functions")

return {
  {
    "folke/snacks.nvim",
    opts = myFunc.SnacksDashboardSettings(),
  },
}
```

When I run Snacks.dashboard.update() nothing happen...  I am able to retrieve a new "setup" (with new content) for dashboard. but I have no idea how to re-render it with that new content.

**Comment by folke** [comment, marked as answer]
What is all this for? `if Snacks and Snacks.dashboard and type(Snacks.dashboard.update) == "function" then`
Please provide a simple example of what you want to achieve.
If you need  parts to re-render, make sure to wrap those sections in functions so they can be re-evaluated.

so `{function() return {text = "foo"} end}` vs `{{text = "foo"}}`

**Comment by Praczet** [comment]
Hi,
`if Snacks and Snacks.dashboard and type(Snacks.dashboard.update) == "function" then`
This is to make sure that snacks is installed and there is function update (this parts works I can Notify it)...

This do the trick:

![image](https://github.com/user-attachments/assets/6c64d6fb-626c-4b80-a2bd-2362a15c5e23)

Thank you!.
I didn't know that this  is possible. I am quite new to all of it.
